### PR TITLE
Convert Exceptions test to spec.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

### CODE REVIEW

File: Library/Homebrew/test/exceptions_spec.rb [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    end
+
+    it "returns a string if there is a dependent" do
+      subject.dependent = "foobar"
+      expect(subject.dependent_s).to eq("(dependency of foobar)")
+    end
+  end
+
+  context "without a dependent" do
+    its(:to_s) { is_expected.to eq("No available formula with the name \"foo\" ") }
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/exceptions_spec.rb]: I'd use `'` when there's a `"` in the string and you don't need interpolation.

File: Library/Homebrew/test/exceptions_spec.rb [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+end
+
+describe NoSuchKegError do
+  subject { described_class.new("foo") }
+  its(:to_s) { is_expected.to eq("No such keg: #{HOMEBREW_CELLAR}/foo") }
+end
+
+describe FormulaValidationError do
+  subject { described_class.new("foo", "sha257", "magic") }
+  its(:to_s) { is_expected.to eq(%q(invalid attribute for formula 'foo': sha257 ("magic"))) }
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/exceptions_spec.rb]: Maybe worth splitting some of these lines that are quite a bit over 80 characters.

File: Library/Homebrew/test/exceptions_spec.rb [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+describe TapFormulaUnavailableError do
+  subject { described_class.new(tap, "foo") }
+  let(:tap) { double(Tap, user: "u", repo: "r", to_s: "u/r", installed?: false) }
+  its(:to_s) { is_expected.to match(%r{Please tap it and then try again: brew tap u/r}) }
+end
+
+describe FormulaClassUnavailableError do
+  subject { described_class.new("foo", "foo.rb", "Foo", list) }
+  let(:mod) do
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/exceptions_spec.rb]: `let(:module)`

Comment (reitermarkus) [inline Library/Homebrew/test/exceptions_spec.rb]: `module` is a keyword.

Comment (MikeMcQuaid) [inline Library/Homebrew/test/exceptions_spec.rb]: Duh. Fine as-is.

**Pull request merged**